tests/rptest/tests/enterprise_features_license_test.py

```diff
@@ -0,0 +1,213 @@
+                                             expect_restart=True)
+        elif feature == Features.partition_auto_balancing_continuous:
+            self.redpanda.set_cluster_config(
+                {'partition_autobalancing_mode': 'continuous'})
+        elif feature == Features.core_balancing_continuous:
+            self.redpanda.set_cluster_config(
+                {'core_balancing_continuous': 'true'})
+        elif feature == Features.gssapi:
+            self.redpanda.set_cluster_config(
+                {'sasl_mechanisms': ['SCRAM', 'GSSAPI']})
+        elif feature == Features.oidc:
+            self.redpanda.set_cluster_config(
+                {'sasl_mechanisms': ['SCRAM', 'OAUTHBEARER']})
+        elif feature == Features.schema_id_validation:
+            self.redpanda.set_cluster_config(
+                {'enable_schema_id_validation': 'compat'})
+        elif feature == Features.rbac:
+
+            # NOTE(oren): make sure the role has propagated to every node since we don't know
+            # where the get_enterprise request will go
+            def has_role(r: str):
+                return all(
+                    len(
+                        RolesList.from_response(
+                            self.admin.list_roles(filter=r, node=n)).roles) > 0
+                    for n in self.redpanda.nodes)
+
+            self.admin.create_role('dummy')
+            wait_until(lambda: has_role('dummy'),
+                       timeout_sec=30,
+                       backoff_sec=1)
+        elif feature == Features.fips:
+            self.redpanda.rolling_restart_nodes(
+                self.redpanda.nodes,
+                override_cfg_params={
+                    'fips_mode':
+                    'permissive',
+                    "openssl_config_file":
+                    self.redpanda.get_openssl_config_file_path(),
+                    "openssl_module_directory":
+                    self.redpanda.get_openssl_modules_directory()
+                })
+        else:
+            assert False, f"Unexpected feature={feature}"
+
+        expect_status = EnterpriseLicenseStatus.valid if with_license else EnterpriseLicenseStatus.not_present
+        rsp = self.admin.get_enterprise_features().json()
+        enabled = {f['name']: f['enabled'] for f in rsp['features']}
+        for f in Features:
+            if f == feature:
+                assert enabled[
+                    f.
+                    name], f"expected {f} enabled, got {json.dumps(enabled, indent=1)}"
+            else:
+                assert not enabled[
+                    f.
+                    name], f"expected {f} not enabled, got {json.dumps(enabled, indent=1)}"
+
+        status = None
+        try:
+            status = EnterpriseLicenseStatus(rsp.get('license_status'))
+        except ValueError:
+            pass
+        assert status == expect_status, f"Unexpected license status: {status}"
```

Comment:
nit: should this print the string from the response just in case status is None?
```suggestion
        assert status == expect_status, f"Unexpected license status: {rsp.get('license_status')}"
```

Reply:
actually I meant to change this to assert in `except ValueError`. good spot though.

Reply:
done